service/service_construction.go

```diff
@@ -89,17 +91,13 @@ func (s ConstructionService) ConstructionMetadata(
 
 	var gasLimit uint64
 	if input.GasLimit == nil {
-		if len(input.To) == 0 || input.Value == nil {
-			// We guard against malformed inputs that may have been generated using
-			// a previous version of avalanche-rosetta.
-			gasLimit = transferGasLimit
+		if input.Currency == nil || types.Hash(input.Currency) == types.Hash(mapper.AvaxCurrency) {
```

Comment:
When would currency be nil here? I guess for ERC-721?

Reply:
Some of the old test cases had a nil Currency field.  If we don't have a currency provided, we assume that it's AVAX (just to make sure we backdate the new functionality)

Reply:
Good find